runtime/lua/vim/lsp/util.lua

```diff
@@ -759,25 +778,26 @@ do
     return popup_bufnr, winnr
   end
 
-  --- Saves the diagnostics (Diagnostic[]) into diagnostics_by_buf
+  --- Saves the diagnostics (Diagnostic[]) into diagnostics_by_buf_and_client_id
   --
-  function M.buf_diagnostics_save_positions(bufnr, diagnostics)
+  function M.buf_diagnostics_save_positions(bufnr, client_id, diagnostics)
```

Comment:
Can you make the new parameter at the end? Also, it should be optional, right? So we can set it to be -1 or whatever  want as a default.

Otherwise this is not backwards compatible as well.